src/jsn.erl

```diff
@@ -656,6 +686,38 @@ from_proplist(X, _Options) -> X.
 %% internal functions
 %%==============================================================================
 
+-spec apply_condition(condition()|conditions(), json_term()) -> boolean().
+apply_condition({Path, ConditionFun}, Element) when is_function(ConditionFun, 1) ->
+    ConditionFun(get(Path, Element));
+apply_condition({Path, Value}, Element) ->
+    get(Path, Element) == Value;
+apply_condition(ConditionFun, Element) when is_function(ConditionFun, 1) ->
+    ConditionFun(Element);
+apply_condition([Condition|Rest], Element) ->
+    case apply_condition(Condition, Element) of
+        true ->
+            apply_condition(Rest, Element);
+        false ->
+            false
+    end;
+apply_condition([], _Element) ->
+    true;
+apply_condition(_, _) ->
+    erlang:error(invalid_condition).
+
+
+-spec apply_selection(selection()|selections(), json_term()) -> term().
+apply_selection(identity, Element) ->
+    Element;
+apply_selection({value, Path}, Element) ->
+    get(Path, Element);
+apply_selection({value, Path, Default}, Element) ->
+    get(Path, Element, Default);
+apply_selection(Selections, Element) when is_list(Selections) ->
+    [apply_selection(Selection, Element) || Selection <- Selections];
```

Comment:
I think it would be more elegant to remove this function clause and explicitly implement an `apply_selections/2` function like:

```erlang
apply_selections(Selections, Element) when is_list(Selections) ->
    [apply_selection(Selection, Element) || Selection <- Selections];
apply_selections(Selection, Element) ->
    apply_selection(Selection, Element).
```

The advantage being, this will not work with selection nesting, e.g. a bizarrely crafted input like

```erlang
[[{value, Path}], [identity], [[{value, Path, Default}, [],[]]]]
```

A construction like that will be more true to the input spec.